File: python/day20.py
presents = [0 for _ in range(1000000)]

for i in range(1, 1000000):
    for j in range(i, 1000000, i):
        presents[j] += i*10
    if presents[i] > 36000000:
        print(i)
        break

presents = [0 for _ in range(1000000)]

for i in range(1, 1000000):
    for j in range(i, min(1000000, 51*i), i):
        presents[j] += i*11
    if presents[i] > 36000000:
        print(i)
        break

File: src/bin/day20.rs
fn main() {
    let target = 36000000;
    let limit = 1000000;
    let mut presents = vec![0; limit];

    for i in 1..limit {
        let mut j = i;
        while j < limit {
            presents[j] += i*10;
            j += i;
        }
    }
    println!("{}", presents.into_iter().position(|p| p > target).unwrap());

    presents = vec![0; limit];

    for i in 1..limit {
        let mut j = i;
        let mut c = 0;
        while j < limit && c < 50 {
            presents[j] += i * 11;
            j += i; c += 1;
        }
    }
    println!("{}", presents.into_iter().position(|p| p > target).unwrap());
}
